src/librbd/Journal.h

```diff
@@ -11,6 +11,7 @@
 #include "common/Cond.h"
 #include "common/Mutex.h"
 #include "common/Cond.h"
+#include "common/WorkQueue.h"
```

Comment:
Please, remove then `class ContextWQ` forward declaration below.

Another approach would be to define Journal<I>::MetadataListener::handle_update() in Journal.cc.